confirm.c

```diff
@@ -763,16 +766,17 @@ confirm_query_create4(const struct sockaddr_storage *dst, uint8_t ttl,
 }
 
 struct confirm_query *
-confirm_query_create6(const struct sockaddr_storage *dst, uint8_t ttl,
-		uint8_t traffic_class, uint32_t flow_label,
-		uint16_t icmpid, uint8_t flowid,
-		confirm_query_cb cb)
+confirm_query_create6_icmp(const struct sockaddr_storage *dst, uint8_t ttl,
+		uint8_t traffic_class, uint32_t flow_label, uint16_t icmpid,
+		uint8_t flowid, confirm_query_cb cb)
 {
-	struct confirm_query *query;
-
-	query = malloc(sizeof(struct confirm_query));
+	struct confirm_query *query = malloc(sizeof(struct confirm_query));
 	if(!query) logea(__FILE__, __LINE__, NULL);
 
+	if(flowid > CONFIRM_MAX_FLOWID) {
+		logd(LOG_WARN, "%s,%d: flowid > 127!\n", __FILE__, __LINE__);
```

Comment:
that might be a good approach (we do use callbacks in various other parts of the code). however, I think what we do not need to make a general library that handles other types of probes. our library is focused on sending icmp and tcp probes to measure routes and router behavior. so, if it would add significant complexity to let the user pass his own functions to the library, there is not much problem in not providing this functionality.